include/seqan3/alignment/matrix/detail/two_dimensional_matrix.hpp

```diff
@@ -337,17 +341,11 @@ class two_dimensional_matrix<value_t, allocator_t, order>::basic_iterator :
     friend class two_dimensional_matrix_iterator_base;
 
     //!\brief Befriend the corresponding const iterator.
-    template <typename other_matrix_t>
-        requires std::same_as<other_matrix_t, std::remove_const_t<matrix_t>> &&
-                    std::is_const_v<matrix_t>
+    template <bool other_const_range>
     friend class basic_iterator;
 
     //!\brief The iterator of the underlying storage.
-    using storage_iterator = std::ranges::iterator_t<
-                                std::conditional_t<
-                                    std::is_const_v<matrix_t>, storage_type const, storage_type
-                                >
-                            >;
+    using storage_iterator = detail::maybe_const_iterator_t<const_range, storage_type>;
```

Comment:
What's this `maybe_const_iterator_t` doing in case that `const_range` is `false`? The name isn't very expressive

Reply:
Yeah this name is not really great. I adapted it from the gcc implementation. 

Iterators who can be const-iterable, s.a. https://eel.is/c++draft/range.transform.iterator, need to specify

```c++
using Parent =                              // exposition only
      conditional_t<Const, const transform_view, transform_view>;

using Base   =                              // exposition only
      conditional_t<Const, const V, V>;
```

so I defined helpers like these:

https://github.com/seqan/seqan3/blob/9d15cbfe4e7d5df422d158102ee52063f86dfc76/include/seqan3/core/type_traits/range.hpp#L37-L50

to give that operation a semantic.

Reply:
ahh ok.. I would really like a more expressive name :see_no_evil: t some point